app/models/targeting.rb

```diff
@@ -0,0 +1,29 @@
+class Targeting < ActiveRecord::Base
+
+  belongs_to :user
+  belongs_to :bookmark
+
+  has_many :targeting_hosts, :dependent => :destroy
+  has_many :hosts, :through => :targeting_hosts
+  has_many :template_invocations, :dependent => :destroy
```

Comment:
That wasn't my impression.  There would be a single targeting and the appropriate template would be used at invocation time for each system.  Specifying a targeting per template seems a tad complex, as I am unlikely to create my bookmarks for a single remote execution provider type.